src/core/core.datasetController.js

```diff
@@ -420,19 +416,21 @@ helpers.extend(DatasetController.prototype, {
 	},
 
 	addElements: function() {
-		var me = this;
-		var meta = me._cachedMeta;
-		var metaData = meta.data;
-		var i, ilen, data;
+		const me = this;
+		const meta = me._cachedMeta;
+		let i, ilen, data;
 
 		me._dataCheck();
 		data = me._data;
+		const metaData = meta.data = new Array(data.length);
 
 		for (i = 0, ilen = data.length; i < ilen; ++i) {
-			metaData[i] = metaData[i] || me.createElement(me.dataElementType);
+			metaData[i] = new me.dataElementType();
```

Comment:
I'm unsure, when the element would have already existed in the old way. Now those ard always create. This might be part of the "shared between multiple charts" stuff

Reply:
Yeah, I changed it because it seemed like an unnecessary extra check. I couldn't figure out when it was ever called and it didn't seem like something we necessarily wanted to keep around, but it'd be good to confirm with @etimberg as well

Reply:
I think it could happen in some weird update scenarios, but I think its OK to create a new one always